app/src/main/java/com/android/example/project2/MoviesFragment.kt

```diff
@@ -37,12 +40,11 @@ class MoviesFragment : Fragment() {
         moviesAdapter = MoviesAdapter(listOf())
         movies.adapter = moviesAdapter
 
-        MoviesRepository.getMovies(
-            onSuccess = { movies ->
-                moviesAdapter.updateMovies(movies)
-        }, onError = {
-            Toast.makeText(requireContext(), "Please check your internet connection", Toast.LENGTH_SHORT).show()
+        moviesViewModel.movies.observe(viewLifecycleOwner, Observer {
+            moviesAdapter.updateMovies(it)
```

Comment:
კონსტრუქოტში არგინდა ოზერვერის მიცემა კიდე. ასეც შეგიძლია :
moviesViewModel.movies.observe(viewLifecycleOwner) {
    moviesAdapter.updateMovies(it)
}

Reply:
ოქეი, მაგრამ ამათ შორის უკეთესი ვარიანტი რატოა ეგ? განსხვავება რა არის..?